### PR TITLE
[INLONG-8958][Manager] Fix the fail for creating cls topic

### LINKED ISSUE #8958: [Bug][Manager]Create cls topic fail
### What happened

when create cls topic,cls sdk throw a exception.Load  to creation failure
企业微信截图_661e0113-5b28-42b7-a9b2-bdcc3f0721cf

### What you expected to happen

create cls resource success

### How to reproduce

create a cls sink

### Environment

_No response_

### InLong version

1.9.0

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by castorqin**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8958 

### Motivation

fix create cls topic fail bug

### Modifications
The details can be found at [#8958 ](https://github.com/apache/inlong/issues/8958).

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 1]
```diff
@@ -55,6 +64,9 @@
 public class ClsResourceOperator extends AbstractStandaloneSinkResourceOperator {
 
     private static final Logger LOG = LoggerFactory.getLogger(ClsResourceOperator.class);
+    private static final String TOPIC_NAME = "topicName";
+    private static final String LOG_SET_ID = "logsetId";
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:68]: private static final String LOG_SET_ID = "logSetId";

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 2]
```diff
@@ -78,32 +90,27 @@ public void createSinkResource(SinkInfo sinkInfo) {
             LOG.warn("create resource was disabled, skip to create for [" + sinkInfo.getId() + "]");
             return;
         }
-        this.createTopicID(sinkInfo);
+        this.createClsResource(sinkInfo);
         this.assignCluster(sinkInfo);
     }
 
     /**
      * Create cloud log service topic
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:98]: createClsResource?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 3]
```diff
@@ -78,32 +90,27 @@ public void createSinkResource(SinkInfo sinkInfo) {
...
-            StreamSinkEntity streamSinkEntity = new StreamSinkEntity();
-            CommonBeanUtils.copyProperties(sinkInfo, streamSinkEntity, true);
-            streamSinkEntityMapper.updateByIdSelective(streamSinkEntity);
+            this.createTopicIndex(sinkInfo, clsDataNode);
+            // update set topic id into sink info
+            updateSinkInfo(sinkInfo, clsSinkDTO);
             String info = "success to create cls resource";
             sinkService.updateStatus(sinkInfo.getId(), SinkStatus.CONFIG_SUCCESSFUL.getCode(), info);
-            LOG.info("update cls sink = {}info status  success ,topicName {}", streamSinkEntity.getSinkName(),
+            LOG.info("update cls sink = {}info status  success ,topicName {}", sinkInfo.getSinkName(),
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:113]: update cls info status  success for sinkId= {},topicName ={}

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 4]
```diff
@@ -113,6 +120,31 @@ private void createTopicID(SinkInfo sinkInfo) {
...
+        }
+        return topicID;
+    }
+
+    private String createTopicReturnTopicId(ClsDataNodeDTO clsDataNode, ClsSinkDTO clsSinkDTO)
+            throws TencentCloudSDKException {
+        ClsClient client = getClsClient(clsDataNode);
+        CreateTopicRequest req = getCreateTopicRequest(clsDataNode, clsSinkDTO);
+        CreateTopicResponse resp = client.CreateTopic(req);
+        LOG.info("create cls topic {} success ,topicId {}", clsSinkDTO.getTopicName(), resp.getTopicId());
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:138]: reate cls topic success for topicName = {}, topicId = {}

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 5]
```diff
@@ -136,14 +168,33 @@ private ClsClient getClsClient(ClsDataNodeDTO clsDataNode) {
     /**
      * Create topic index by tokenizer
      */
-    private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
+    private void createTopicIndex(SinkInfo sinkInfo, ClsDataNodeDTO clsDataNode) throws BusinessException {
+
         ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
-        if (StringUtils.isNotBlank(clsSinkDTO.getTokenizer())) {
+        if (StringUtils.isEmpty(clsSinkDTO.getTokenizer())) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:174]: StringUtils.isBlank

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 6]
```diff
@@ -136,14 +168,33 @@ private ClsClient getClsClient(ClsDataNodeDTO clsDataNode) {
...
+        CreateIndexRequest req = getCreateIndexRequest(clsSinkDTO);
+        try {
+            clsClient.CreateIndex(req);
+        } catch (TencentCloudSDKException e) {
+            String errMsg = "Create cls topic index failed: " + e.getMessage();
+            LOG.error(errMsg, e);
+            sinkService.updateStatus(sinkInfo.getId(), SinkStatus.CONFIG_FAILED.getCode(), errMsg);
+            throw new BusinessException(errMsg);
+        }
+        LOG.info("topic {} create index success tokenizer is {}", clsSinkDTO.getTopicName(), clsSinkDTO.getTokenizer());
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:194]: create index success for topic ={}, tokenizer = {}

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 7]
```diff
@@ -152,15 +203,86 @@ private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
...
+        }
+    }
+
+    private Filter[] getDescribeFilters(ClsDataNodeDTO clsDataNode, ClsSinkDTO clsSinkDTO) {
+        Filter filter = new Filter();
+        String[] filterValues = new String[1];
+        filterValues[0] = clsSinkDTO.getTopicName();
+        filter.setKey(TOPIC_NAME);
+        filter.setValues(filterValues);
+        Filter filter1 = new Filter();
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:281]: Filter filter1?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 8]
```diff
@@ -152,15 +203,86 @@ private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
...
+        String[] filterValues1 = new String[]{clsDataNode.getLogSetId()};
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:282]: String[] filterValues1?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/AbstractSinkOperator.java [hunk 9]
```diff
@@ -218,14 +219,19 @@ public void deleteOpt(StreamSinkEntity entity, String operator) {
...
-            assert param != null;
+            HashMap<String, Object> streamInfoMap = JsonUtils.parseObject(streamSink.getExtParams(), HashMap.class);
+            param = new HashMap<>();
+            assert streamInfoMap != null;
+            for (String key : streamInfoMap.keySet()) {
+                param.put(key, String.valueOf(streamInfoMap.get(key)));
+            }
             param.put(KEY_GROUP_ID, streamSink.getInlongGroupId());
             param.put(KEY_STREAM_ID, streamSink.getInlongStreamId());
+            // put group and stream info
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/AbstractSinkOperator.java:234]:
```suggestion
            // put group and stream info
            param.put(KEY_GROUP_ID, streamSink.getInlongGroupId());
            param.put(KEY_STREAM_ID, streamSink.getInlongStreamId());
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 7 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:281]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 8 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:282]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 6 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:194]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 5 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:174]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 3 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:113]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 4 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:138]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 1 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:68]: tencent cloud log service sdk query  filters by logsetId

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 2 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:98]: create topic ,and create topic index

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/AbstractSinkOperator.java (hunk 9 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/AbstractSinkOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 10]
```diff
@@ -78,32 +90,27 @@ public void createSinkResource(SinkInfo sinkInfo) {
...
         ClsDataNodeDTO clsDataNode = getClsDataNode(sinkInfo);
         ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
         try {
-            ClsClient client = getClsClient(clsDataNode);
-            CreateTopicRequest req = getCreateTopicRequest(clsDataNode, clsSinkDTO);
-            CreateTopicResponse resp = client.CreateTopic(req);
-            LOG.info("create cls topic {} success ,topicId {}", clsSinkDTO.getTopicName(), resp.getTopicId());
-            // update set topic id into sink info
-            clsSinkDTO.setTopicId(resp.getTopicId());
+            String topicID = getTopicID(sinkInfo, clsDataNode, clsSinkDTO);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:104]: topicId

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 11]
```diff
@@ -78,32 +90,27 @@ public void createSinkResource(SinkInfo sinkInfo) {
...
-            StreamSinkEntity streamSinkEntity = new StreamSinkEntity();
-            CommonBeanUtils.copyProperties(sinkInfo, streamSinkEntity, true);
-            streamSinkEntityMapper.updateByIdSelective(streamSinkEntity);
+            this.createTopicIndex(sinkInfo, clsDataNode);
+            // update set topic id into sink info
+            updateSinkInfo(sinkInfo, clsSinkDTO);
             String info = "success to create cls resource";
             sinkService.updateStatus(sinkInfo.getId(), SinkStatus.CONFIG_SUCCESSFUL.getCode(), info);
-            LOG.info("update cls sink = {}info status  success ,topicName {}", streamSinkEntity.getSinkName(),
+            LOG.info("update cls info status success for sinkId= {},topicName ={}", sinkInfo.getSinkName(),
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:113]: for sinkId = {}, topicName = {}

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 12]
```diff
@@ -113,6 +120,32 @@ private void createTopicID(SinkInfo sinkInfo) {
         }
     }
 
+    private String getTopicID(SinkInfo sinkInfo, ClsDataNodeDTO clsDataNode, ClsSinkDTO clsSinkDTO)
+            throws TencentCloudSDKException {
+        String topicID = describeTopicIDByTopicName(sinkInfo, clsDataNode);
+        if (Strings.isEmpty(topicID)) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:126]: Strings.isBlank(topicID)

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 13]
```diff
@@ -136,14 +169,34 @@ private ClsClient getClsClient(ClsDataNodeDTO clsDataNode) {
...
      * Create topic index by tokenizer
      */
-    private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
+    private void createTopicIndex(SinkInfo sinkInfo, ClsDataNodeDTO clsDataNode) throws BusinessException {
+
         ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
-        if (StringUtils.isNotBlank(clsSinkDTO.getTokenizer())) {
-            LOG.warn("topic {} tokenizer is empty", clsSinkDTO.getTopicName());
+        if (StringUtils.isBlank(clsSinkDTO.getTokenizer())) {
+            LOG.warn("topic {} tokenizer is blank", clsSinkDTO.getTopicName());
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:176]: tokenizer is blank for topic = {}

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 14]
```diff
@@ -136,14 +169,34 @@ private ClsClient getClsClient(ClsDataNodeDTO clsDataNode) {
...
+        CreateIndexRequest req = getCreateIndexRequest(clsSinkDTO);
+        try {
+            clsClient.CreateIndex(req);
+        } catch (TencentCloudSDKException e) {
+            String errMsg = "Create cls topic index failed: " + e.getMessage();
+            LOG.error(errMsg, e);
+            sinkService.updateStatus(sinkInfo.getId(), SinkStatus.CONFIG_FAILED.getCode(), errMsg);
+            throw new BusinessException(errMsg);
+        }
+        LOG.info("create index success for topic ={}, tokenizer = {}", clsSinkDTO.getTopicName(),
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:195]: create index success for topic = {}, tokenizer = {}

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 15]
```diff
@@ -152,15 +205,87 @@ private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
...
         req.setTopicId(clsSinkDTO.getTopicId());
         req.setRule(ruleInfo);
+        return req;
+    }
+
+    private void updateTopicIndex(SinkInfo sinkInfo, ClsDataNodeDTO clsDataNode) {
+        ClsClient clsClient = getClsClient(clsDataNode);
+        ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
+        RuleInfo ruleInfo = new RuleInfo();
+        FullTextInfo fullTextInfo1 = new FullTextInfo();
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:215]: FullTextInfo fullTextInfo1?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 16]
```diff
@@ -152,15 +205,87 @@ private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
...
+    private String describeTopicIDByTopicName(SinkInfo sinkInfo, ClsDataNodeDTO clsDataNode) {
+        ClsClient clsClient = getClsClient(clsDataNode);
+        ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
+        Filter[] filters = getDescribeFilters(clsDataNode, clsSinkDTO);
+        DescribeTopicsRequest req = new DescribeTopicsRequest();
+        req.setFilters(filters);
+        req.setPreciseSearch(PRECISE_SEARCH);
+        try {
+            DescribeTopicsResponse describeTopicsResponse = clsClient.DescribeTopics(req);
+            LOG.info("sink {} describe cls topic success topic count {}", sinkInfo.getSinkName(),
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:244]: success describe cls topic count = {} for sinkId = {}

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 17]
```diff
@@ -169,7 +294,8 @@ private Tag[] convertTags(String[] allTags) {
             String tag = allTags[i];
             String[] keyAndValueOfTag = tag.split(InlongConstants.COLON);
             Tag tagInfo = new Tag();
-            tagInfo.set(keyAndValueOfTag[0], keyAndValueOfTag[1]);
+            tagInfo.setKey(keyAndValueOfTag[0]);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:297]: Maybe `IndexOutOfBounds`?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 18]
```diff
@@ -78,32 +90,27 @@ public void createSinkResource(SinkInfo sinkInfo) {
...
+            this.createTopicIndex(sinkInfo, clsDataNode);
+            // update set topic id into sink info
+            updateSinkInfo(sinkInfo, clsSinkDTO);
             String info = "success to create cls resource";
             sinkService.updateStatus(sinkInfo.getId(), SinkStatus.CONFIG_SUCCESSFUL.getCode(), info);
-            LOG.info("update cls sink = {}info status  success ,topicName {}", streamSinkEntity.getSinkName(),
+            LOG.info("update cls info status success for sinkId= {},topicName ={}", sinkInfo.getSinkName(),
                     clsSinkDTO.getTopicName());
         } catch (TencentCloudSDKException e) {
             String errMsg = "Create cls topic  failed: " + e.getMessage();
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:116]: Create cls topic failed:

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 19]
```diff
@@ -152,15 +205,87 @@ private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
...
-            clsClient.CreateIndex(req);
+            clsClient.ModifyIndex(req);
         } catch (TencentCloudSDKException e) {
-            String errMsg = "Create cls topic index failed: " + e.getMessage();
+            String errMsg = "update cls topic index failed: " + e.getMessage();
             LOG.error(errMsg, e);
             sinkService.updateStatus(sinkInfo.getId(), SinkStatus.CONFIG_FAILED.getCode(), errMsg);
             throw new BusinessException(errMsg);
         }
-        LOG.info("topic {} create index success tokenizer is {}", clsSinkDTO.getTopicName(), clsSinkDTO.getTokenizer());
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:163]: LOG.info("update index success for topic = {}, tokenizer = {}", clsSinkDTO.getTopicName(),,clsSinkDTO.getTokenizer());

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 20]
```diff
@@ -136,14 +169,34 @@ private ClsClient getClsClient(ClsDataNodeDTO clsDataNode) {
...
+        if (ObjectUtils.anyNotNull(topicIndexFullText)) {
+            // if topic index exist,update
+            updateTopicIndex(sinkInfo, clsDataNode);
             return;
         }
-        ClsDataNodeDTO clsDataNode = getClsDataNode(sinkInfo);
         ClsClient clsClient = getClsClient(clsDataNode);
+        CreateIndexRequest req = getCreateIndexRequest(clsSinkDTO);
+        try {
+            clsClient.CreateIndex(req);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:188]: Please add log for `CreateIndexResponse`.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 21]
```diff
@@ -152,15 +205,87 @@ private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
...
+        FullTextInfo fullTextInfo1 = new FullTextInfo();
+        fullTextInfo1.setTokenizer(clsSinkDTO.getTokenizer());
+        ruleInfo.setFullText(fullTextInfo1);
+
+        ModifyIndexRequest req = new ModifyIndexRequest();
+        req.setTopicId(clsSinkDTO.getTopicId());
+        req.setRule(ruleInfo);
         try {
-            clsClient.CreateIndex(req);
+            clsClient.ModifyIndex(req);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:223]: Please add log for `ModifyIndexResponse`.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 22]
```diff
@@ -113,6 +120,32 @@ private void createTopicID(SinkInfo sinkInfo) {
         }
     }
 
+    private String getTopicID(SinkInfo sinkInfo, ClsDataNodeDTO clsDataNode, ClsSinkDTO clsSinkDTO)
+            throws TencentCloudSDKException {
+        String topicID = describeTopicIDByTopicName(sinkInfo, clsDataNode);
+        if (Strings.isEmpty(topicID)) {
+            // if topic don't exist,create topic in cls
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:127]: if topic don't exist, create topic in cls

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java [hunk 23]
```diff
@@ -136,14 +169,34 @@ private ClsClient getClsClient(ClsDataNodeDTO clsDataNode) {
...
         ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
-        if (StringUtils.isNotBlank(clsSinkDTO.getTokenizer())) {
-            LOG.warn("topic {} tokenizer is empty", clsSinkDTO.getTopicName());
+        if (StringUtils.isBlank(clsSinkDTO.getTokenizer())) {
+            LOG.warn("topic {} tokenizer is blank", clsSinkDTO.getTopicName());
+            return;
+        }
+        FullTextInfo topicIndexFullText = getTopicIndexFullText(sinkInfo, clsDataNode);
+        if (ObjectUtils.anyNotNull(topicIndexFullText)) {
+            // if topic index exist,update
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java:181]: if topic index exist, update

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 11 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 12 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 10 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 14 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 16 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 17 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 15 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 18 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 21 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 23 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 22 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 20 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java (hunk 13 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java]: done

**Review by fuweng11 — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Pull request merged**